src/fides/api/main.py

```diff
@@ -50,6 +51,21 @@
 
 app = create_fides_app()
 
+if CONFIG.dev_mode:
+
+    @app.middleware("http")
+    async def profile_request(request: Request, call_next: Callable) -> Response:
+        profiling = request.query_params.get("profile", False)
```

Comment:
i really like this middleware design, we get a big bang for our buck: super lightweight but very powerful.

one nit: could this be a request header rather than query param? i don't have many concrete reasons to suggest that, more of just a gut feeling. if i searched my gut (gross) for reasons, here's what i found: 
- i usually think of a query-param as parameterization local to a given endpoint and the endpoint's specific business logic, whereas headers typically hold general metadata that's more common throughout all (or most) of the endpoints in a given API? 
- i think there's also a _slight_ chance that a param like `profile` could actually be used for another purpose by a specific API at some point, whereas a `profile-request` header feels a lot less ripe for ambiguity or overloading.

if you feel convinced that the query param here is the better option, then i'll defer to you - i don't feel too strongly about this!

```suggestion
        profiling = request.headers.get("profile-request", False)
```

Reply:
More of a smart design decision than a nit 🙂 good call